File: minDeletionSize.py
# Approach 1- Simple two-for loops
def minDeletionSize(strs): # TC O(m*n) // SC O(1)
	wordLen=len(strs[0])
	n=len(strs)
	delete=0
	for i in range(wordLen):
	    for j in range(n-1):
	        if strs[j][i]>strs[j+1][i]:
	            delete+=1 
	            break 

	return delete 

print(minDeletionSize(["cba","daf","ghi"]))

File: minDeletionSize.java
import java.util.*;
// Approach 1- Simple two-for loops 
class Solution { // TC O(m*n) SC O(1)
    public int minDeletionSize(String[] strs) {
        int del=0;
        int n=strs[0].length();
        for (int j=0; j<n; j++){
            for(int i=0; i<strs.length-1; i++){
                if(strs[i].charAt(j)>strs[i+1].charAt(j)){
                    del++;
                    break;
                }
            }
        }
        return del;
    }
}

class minDeletionSize{
    public static void main(String[] args) {
        String[] strs={"cba","daf","ghi"};
        Solution s= new Solution();
        System.out.println(s.minDeletionSize(strs));
    }
}